rs/log-fetcher/src/journald_parser.rs

```diff
@@ -12,7 +12,81 @@ pub struct JournalEntry {
     pub fields: Vec<(String, JournalField)>,
 }
 
-pub fn parse_journal_entries(body: &[u8]) -> Vec<JournalEntry> {
+#[derive(Debug, Clone)]
+enum LineStatus {
+    NotStarted,
+    Started,
+    Utf8,
+    Binary,
+}
+
+pub fn parse_journal_entries_new(body: &[u8]) -> Vec<JournalEntry> {
+    let mut entries = Vec::new();
+    let mut current_entry = Vec::new();
+    let mut current_line = Vec::new();
+
+    let mut first_found = LineStatus::NotStarted;
+
+    let mut iter = body.iter();
+    while let Some(byte) = iter.next() {
```

Comment:
Some comment on how the state machine / input parsing works on a high level would be awesome.